### PR TITLE
glooctl: Improve performance of the check command by optimizing the creation of clients 

### PR DESCRIPTION
**Pull request by davidjumani**
# Description

Optimize `glooctl check` by creating the resource client prior to iterating over namespaces.
This saves a considerable amount of time in large envs consisting of over 700 ns and cuts time by ~50%

# Context

See slack conversation [here](https://solo-io-corp.slack.com/archives/CEDCS8TAP/p1718935489580349)

## Testing steps
In an env of 700+ ns

glooctl version: `1.17.0-beta34(latest version)`
```
glooctl check  251.22s user 266.20s system 37% cpu 22:48.24 total
```
glooctl version: `1.0.0-ci1 (This PR)`
```
glooctl-darwin-arm64-david check  81.22s user 74.02s system 24% cpu 10:28.20 total
```

# Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works

BOT NOTES: 
resolves https://github.com/solo-io/gloo/issues/9673

### CODE REVIEW

File: projects/gloo/cli/pkg/cmd/check/root.go [hunk 1]
```diff
@@ -425,8 +425,8 @@ func checkUpstreams(ctx context.Context, printer printers.P, _ *options.Options,
 	printer.AppendCheck("Checking Upstreams... ")
 	var knownUpstreams []string
 	var multiErr *multierror.Error
+	client, err := helpers.UpstreamClient(ctx, namespaces)
```
Comment (nfuden) [inline projects/gloo/cli/pkg/cmd/check/root.go:428]: Can you please test with multiple namespaces and check that no duplicates occur?

File: projects/gloo/cli/pkg/cmd/check/root.go [hunk 2]
```diff
@@ -425,8 +425,8 @@ func checkUpstreams(ctx context.Context, printer printers.P, _ *options.Options,
...
 	for _, ns := range namespaces {
-		client, err := helpers.UpstreamClient(ctx, []string{ns})
```
Comment (nfuden) [inline projects/gloo/cli/pkg/cmd/check/root.go:429]: Without this line that is scoped to name space what is the value of running list per namespace while having the whole list of namespaces?

File: projects/gloo/cli/pkg/cmd/check/root.go (hunk 1 above)
Comment (davidjumani) [reply projects/gloo/cli/pkg/cmd/check/root.go:428]: Manually tested and believe [this test](https://github.com/solo-io/gloo/blob/b76819b53fe310e3dc0d5f909172eed9344da84d/test/kube2e/glooctl/check_test.go#L391) covers it

File: projects/gloo/cli/pkg/cmd/check/root.go (hunk 2 above)
Comment (davidjumani) [reply projects/gloo/cli/pkg/cmd/check/root.go:429]: It is just for performance. It avoids creating the client scoped to just the namespace during the iteration but scoped to all namespaces to check

**Review by jenshu — APPROVED** [review]

**Review by nfuden — APPROVED** [review]

**Comment by jenshu** [comment]
merge

**Pull request merged**